examples/templates/azure-windows/FirstLogonCommands.xml

```diff
@@ -5,8 +5,8 @@
 		<Order>10</Order>
 	</SynchronousCommand>
 	<SynchronousCommand>
-		<CommandLine>powershell.exe -sta -ExecutionPolicy Unrestricted -Command "C:\AzureData\CoderAgent.ps1 | Out-File C:\AzureData\CoderAgent.log</CommandLine>
-		<Description>Execute CustomData as powershell.</Description>
+		<CommandLine>powershell.exe -ExecutionPolicy Unrestricted -Command "Register-ScheduledTask -TaskName 'MyStartupScript' -Action (New-ScheduledTaskAction -Execute 'powershell.exe' -Argument '-sta -ExecutionPolicy Unrestricted -Command `"C:\AzureData\CoderAgent.ps1 ^| Out-File C:\AzureData\CoderAgent.log`"') -Trigger (New-ScheduledTaskTrigger -AtStartup), (New-ScheduledTaskTrigger -Once -At (Get-Date).AddSeconds(15)) -Settings (New-ScheduledTaskSettingsSet -DontStopOnIdleEnd -ExecutionTimeLimit ([TimeSpan]::FromDays(3650)) -Compatibility Win8) -Principal (New-ScheduledTaskPrincipal -UserId 'vm\coder' -RunLevel Highest -LogonType S4U) -Force"</CommandLine>
+		<Description>Register the scheduled task</Description>
```

Comment:
This works, but I'm actually thinking that registering the Coder Agent as a Service is the more Windows-y thing to do.  E.g. if it's just run as a Scheduled Task, and you RDP into the box, I think the Coder Agent is just like running in a command window on your desktop.  It would be really easy to not know what it is, and close it, bricking your workspace until you rebuild.

Windows Services are a unified system for running background processes, sort of like `systemd` on Linux.

Reply:
FWIW when I connect with RDP, I don't see powershell open. I think `-LogonType S4U` contributes to this.

Down to make a more appropriate approach though, such as Windows services. Is this something you could tackle?

Reply:
Oh, that's helpful.

Might still be nice to do a proper Service at some point down the road, but less pressing if you don't see the Agent in the foreground when you RDP.